tests/testsuite.src/run_misc.at

```diff
@@ -3969,6 +3969,38 @@ AT_CHECK([$COBCRUN_DIRECT ./ebcdic], [0], [], [])
 AT_CLEANUP
 
 
+AT_SETUP([SEARCH ALL with non-0xff HIGH-VALUE])
+AT_KEYWORDS([runmisc default-colseq])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01  TAB.
+         02 TAB-ELT   OCCURS 3
+                      ASCENDING KEY TAB-KEY
+                      INDEXED BY TI.
+           05 TAB-KEY PIC X.
+       PROCEDURE DIVISION.
+           MOVE HIGH-VALUE TO TAB
+           MOVE "1" TO TAB-ELT (1)
+      *    DISPLAY "|" TAB "|"
+           SEARCH ALL TAB-ELT
+              AT END
+                 DISPLAY '"1" NOT FOUND'
+              WHEN TAB-KEY (TI) = "1"
+                 CONTINUE
+           END-SEARCH
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE -febcdic-table=ebcdic500_latin1 -fdefault-colseq=EBCDIC prog.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [], [])
+
+AT_CLEANUP
```

Comment:
@GitMensch This origins from a coding pattern we've witnessed in a code base. Without the changes in this PR the search fails. Even if that requires a rather unusual collating sequence that leads to `HIGH-VALUE` not being `0xff`, I think that's one more case for considering an upstream of this PR.

Reply:
> Even if that requires a rather unusual collating sequence 

Not so unsual IMHO (EBCDIC-500 / Latin-1).